packages/foundry-sdk-generator/src/generate/GeneratePackageCommand.test.ts

```diff
@@ -48,4 +50,109 @@ describe(GeneratePackageCommand, () => {
     expect(existsSync(typesPath), typesPath).toBe(true);
     expect(existsSync(esmPath), esmPath).toBe(true);
   });
+
+  it("has a .d.ts for slate that does not error", () => {
+    const slateDtsPath = path.join(
+      __dirname,
+      "..",
+      "generatedNoCheck",
+      "@test-app",
+      "osdk",
+      "dist",
+      "bundle",
+      "index.d.ts",
+    );
+
+    const compilerOptions: CompilerOptions = {
+      module: ts.ModuleKind.CommonJS,
+      target: ts.ScriptTarget.ES2019,
+      noEmit: true,
+      skipLibCheck: false,
+    };
+    const compilerHost = ts.createCompilerHost(compilerOptions);
+
+    const fauxProjectDir = "/not-real";
+    const notRealFilePath = path.join(fauxProjectDir, "index.d.ts");
+
+    const ogGetSourceFile = compilerHost.getSourceFile;
+    compilerHost.getSourceFile = (
+      fileName: string,
+      languageVersionOrOptions: ts.ScriptTarget | ts.CreateSourceFileOptions,
+      onError?: (message: string) => void,
+      shouldCreateNewSourceFile?: boolean,
+    ): ts.SourceFile | undefined => {
+      if (fileName === notRealFilePath) {
+        const realContents = compilerHost.readFile(slateDtsPath)!;
+
+        // We are going to introduce a fake error to ensure that the type checking is working correctly
+        // and then we can expect only the fake error to be in the diagnostics
+        const contentsWithFakeError = realContents
+          + `\ndeclare module "asdf-asdf-asdf-asdf" { import { notARealThing} from "internal/@osdk/api"; }`;
+
+        return ts.createSourceFile(
+          notRealFilePath,
+          contentsWithFakeError,
+          ts.ScriptTarget.ES2019,
+        );
+      }
+
+      if (fileName.includes("node_modules/typescript/lib")) {
+        return ogGetSourceFile(
+          fileName,
+          languageVersionOrOptions,
+          onError,
+          shouldCreateNewSourceFile,
+        );
```

Comment:
Can you explain why we need to do this

Reply:
A future change could turn of lib checking. Or add a non existent file which has no diagnostics for some reason. This is just a clever defensive trick